src/main/java/teammates/sqllogic/core/UsersLogic.java

```diff
@@ -740,34 +746,55 @@ private boolean isSectionChanged(Section originalSection, Section newSection) {
     public Student updateStudentCascade(Student student)
             throws InvalidParametersException, EntityDoesNotExistException, EntityAlreadyExistsException {
 
-        Student originalStudent = getStudentForEmail(student.getCourseId(), student.getEmail());
+        String courseId = student.getCourseId();
+        Student originalStudent = getStudent(student.getId());
+        String originalEmail = originalStudent.getEmail();
+        boolean changedEmail = isEmailChanged(originalEmail, student.getEmail());
+
+        // check for email conflict
+        Student s = usersDb.getStudentForEmail(courseId, student.getEmail());
+        if (changedEmail && s != null) {
+            String errorMessage = String.format(ERROR_CREATE_ENTITY_ALREADY_EXISTS, s.toString());
+            throw new EntityAlreadyExistsException(errorMessage);
+        }
+
         Team originalTeam = originalStudent.getTeam();
         Section originalSection = originalStudent.getSection();
-
         boolean changedTeam = isTeamChanged(originalTeam, student.getTeam());
         boolean changedSection = isSectionChanged(originalSection, student.getSection());
 
+        // update student
+        usersDb.checkBeforeUpdateStudent(student);
         originalStudent.setName(student.getName());
         originalStudent.setTeam(student.getTeam());
         originalStudent.setEmail(student.getEmail());
         originalStudent.setComments(student.getComments());
 
-        Student updatedStudent = usersDb.updateStudent(originalStudent);
-        Course course = updatedStudent.getCourse();
+        // cascade email changes to account, responses and comments
+        if (changedEmail) {
+            feedbackResponsesLogic.updateFeedbackResponsesForChangingEmail(courseId, originalEmail, student.getEmail());
+            feedbackResponseCommentsLogic.updateFeedbackResponseCommentsEmails(courseId, originalEmail, student.getEmail());
+
+            Account originalAccount = originalStudent.getAccount();
```

Comment:
Similar to the previous comment, I think Account and Student emails can be different therefore, the email does not have to be cascaded to account. @wkurniawan07 is that correct?

Reply:
Looking at the original code I think this is confused with googleid, which from my understanding was not a field that was brought over to the postgres entity, therefore this cascading is not needed @cedricongjh @weiquu

Reply:
Not only it doesn't have to be cascaded, it must not be cascaded.